pubsub/cli.rb

```diff
@@ -0,0 +1,127 @@
+#!/usr/bin/env ruby
+# Copyright 2015 Google, Inc
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require "gcloud"
+
+# Shows using gcloud to interact with Pub/Sub
+class PubSubSample
+  attr_accessor :topic, :sub
+
+  def initialize project
+    @project = project
+    @topic = "ruby-sample"
+    @sub = "ruby-sample-sub"
+    @subp = "ruby-sample-sub-p"
+
+    gcloud = Gcloud.new @project
+    pubsub = gcloud.pubsub
+
+    topic = pubsub.topic @topic, autocreate: true
+
+    sub = topic.get_subscription @sub
+    topic.subscribe @sub if sub.nil?
+
+    subp = topic.get_subscription @subp
+    topic.subscribe(
+      @subp,
+      endpoint: "https://#{@project}.appspot.com/push") if subp.nil?
+  end
+
+  def publish message
+    gcloud = Gcloud.new @project
+    pubsub = gcloud.pubsub
+    topic = pubsub.topic @topic
```

Comment:
In Ruby we very frequently line up our assignments `=`

*Especially* for code samples.

```ruby
gcloud = 1
foo    = 2
hello  = 3
```

Also, let's add a newline before the "action" operations for each code sample:

```ruby
def pull
  # setup for this sample
  gcloud       = Gcloud.new @project
  pubsub       = gcloud.pubsub
  subscription = pubsub.subscription @subscription_name

  # action this sample demonstrates
  subscription.pull.each do |message|
    puts message.data
  end
end
```